feat(trends): add a section for trends (#2556)
getsentry/sentry-docs: src/docs/product/performance/index.mdx

## Key Transaction View

If you have transactions you frequently return to; you can mark each one as a key transaction in their corresponding [summary views](/product/performance/transaction-summary/#key-transaction). This enables a shortcut to view only Key Transactions by toggling the tab in the upper right corner of the homepage.

## Trends View

> Trends surfaces transactions that have had significant changes in their performance over time. This view can be found by toggling the tab in the upper right corner of the homepage.

Reviewer: Unsure about `This view can be found by toggling the tab in the upper right corner of the homepage.` People might get confused with `homepage`.

`You can find it by going to the top right corner beside Transactions and Key Transactions.`

> ![trends Homepage](trends-homepage.png)

Reviewer: The screenshot is missing the key transactions tab

Author: :+1: will keep this comment unresolved as a reminder to update this later.

Author: So I've updated this to the latest, to save some work I haven't pushed to put the key transaction tab back in since it'll be going away fairly soon

Trending transactions are calculated first by filtering out transactions with large fluctuations in [throughput](/product/performance/metrics/#throughput-total-tpm-tps), then determining the baselines of the two halves of the duration, and calculating their percentage change. Regressions are any percentage changes greater than 0, while improvements are percentage changes less than 0.

> For example looking at a transaction over two weeks with the default [P50](/product/performance/metrics/#average-transaction-duration) baseline, if the first week had a `p(50)=100s`, and the second a `p(50)=110s` the percentage is `(110/100)-1 = 0.1`. This means that this example transaction had a performance regression of `10%`.

Reviewer: `For example, ...`

The accuracy of a trend improves the more data that is available, for this reason, trends perform best on long periods, and with transactions with large counts.

> While the baseline defaults to P50 but the Display dropdown lets you change this. The baseline is used to determine the percentage change but is also used for the comparison view found in the dropdown for each transaction. This comparison view finds the transaction in each period with the duration closest to the baseline and compares their spans.

Reviewer: The first sentence is a little confusing.

`While we default the baseline calculation to use P50 you can change it via the dropdown, other options include P75, P95, P99 and average.`

## Key Transaction View

If you have transactions you frequently return to; you can mark each one as a key transaction in their corresponding [summary views](/product/performance/transaction-summary/#key-transaction). This enables a shortcut to view only Key Transactions by toggling the tab in the upper right corner of the homepage.

## Trends View

Trends surfaces transactions that have had significant changes in their performance over time. This view can be found by toggling the tab in the upper right corner of the homepage.

![trends Homepage](trends-homepage.png)

> Trending transactions are calculated first by filtering out transactions with large fluctuations in [throughput](/product/performance/metrics/#throughput-total-tpm-tps), then determining the baselines of the two halves of the duration, and calculating their percentage change. Regressions are any percentage changes greater than 0, while improvements are percentage changes less than 0.

Reviewer 2: Do we need to update this because of any changes to the logic for throughput filtering? IIRC we talked about potentially relaxing the requirements for the ratio of before/after volume (and it may be interesting to surface these changes to the customer even if transaction speed doesn't change much), especially since we can now use a t-test to throw out stuff that's not interesting.

Author: In the short-term we won't be surfacing throughput changes on the trends view, and for now we'll have the throughput filtering so I don't think this needs an update (but I'll keep this comment unresolved as a reminder to update this if we do remove the throughput filtering)

> The default view for trends will only show high confidence trends, these are based on transactions that have a high [t-test](https://en.wikipedia.org/wiki/Welch%27s_t-test). High magnitude t-tests being used as a signal that there has been a significant change between periods.

Reviewer 2: What about something like this?
```suggestion
The `Confidence` option allows you to filter trends based on High (trend is statistically significant) or Low (trend is not statistically significant) significance. By default, only high confidence, statistically significant results are shown. Sentry uses a [t-test](https://en.wikipedia.org/wiki/Welch%27s_t-test) to compare the before and after periods of the transaction and assign a confidence score. 
```

Author: Hmm what about this? The bits in the brackets seem a bit redundant to me
```suggestion
The `Confidence` option allows you to filter trends based on High or Low statistical significance. By default, only high confidence, statistically significant results are shown. Sentry uses a [t-test](https://en.wikipedia.org/wiki/Welch%27s_t-test) to compare the before and after periods of the transaction and assign a confidence score. 
```

Reviewer 2: ya agreed that's better

> The `Confidence` option allows you to filter trends based on High or Low statistical significance. By default, only high confidence, statistically significant results are shown. Sentry uses a [t-test](https://en.wikipedia.org/wiki/Welch%27s_t-test) to compare the before and after periods of the transaction and assign a confidence score. 

Author: Hmm using something more reputable thank Wikipedia is a great idea, the Berkeley one looks to be about how to calculate it using R, what about this Oxford link?
https://academic.oup.com/beheco/article/17/4/688/215960

(cc @trillville in case you have a better suggestion)

> The `Confidence` option allows you to filter trends based on High or Low statistical significance. By default, only high confidence, statistically significant results are shown. Sentry uses a [t-test](https://en.wikipedia.org/wiki/Welch%27s_t-test) to compare the before and after periods of the transaction and assign a confidence score. 

Reviewer 2: No strong feelings from me - imo the wikipedia link is more "useful" in the sense that it's more likely that it's more likely to actually provide context to someone that's interested in learning more but doesn't want to necessarily dive into academic literature, so I'd probably vote to just use that personally.

> For example, looking at a transaction over two weeks with the default [P50](/product/performance/metrics/#average-transaction-duration) baseline, if the first week had a `p(50)=100s`, and the second a `p(50)=110s` the percentage is `(110/100)-1 = 0.1`. This means that this example transaction had a performance regression of `10%`.

Reviewer 3: ```suggestion
For example, looking at a transaction over two weeks with the default [P50](/product/performance/metrics/#average-transaction-duration) baseline, if the first week had a `p(50)=100s`, and the second a `p(50)=110s`, the percentage is `(110/100)-1 = 0.1`. The example transaction had a performance regression of `10%`.
```

> The accuracy of a trend improves the more data that is available, for this reason, trends perform best on long periods, and with transactions with large counts.

Reviewer 3: ```suggestion
The more data is available, the more the accuracy of the trend improves. For this reason, trends perform best over long periods and with transactions with large counts.
```

> While we default the baseline calculation to use P50 you can change it via the dropdown, other options include P75, P95, P99 and average. The baseline is used to determine the percentage change but is also used for the comparison view found in the dropdown for each transaction. This comparison view finds the transaction in each period with the duration closest to the baseline and compares their spans.

Reviewer 3: ```suggestion
While we default the baseline calculation to use P50, you can change the baseline calculation via the dropdown. Other options include P75, P95, P99, and average. The baseline determines the percentage change, but you can also use it for the comparison view found in each transaction's dropdown. The comparison view finds the transaction in each period with the duration closest to the baseline and compares their spans.
```

> While we default the baseline calculation to use P50, you can change the baseline calculation via the dropdown. Other options include P75, P95, P99, and average. The baseline determines the percentage change, but you can also use it for the comparison view found in each transaction's dropdown. The comparison view finds the transaction in each period with the duration closest to the baseline and compares their spans.

Reviewer 4: `The baseline determines the percentage change, but you can also use it for the comparison view found in each transaction's dropdown.` we're taking out comparison view for GA.
